src/libs/actions/Report.js

```diff
@@ -660,7 +661,7 @@ function handleReportChanged(report) {
     // A report can be missing a name if a comment is received via pusher event
     // and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
     if (report.reportID && report.reportName === undefined) {
-        openReport(report.reportID);
+        reconnectToReport(report.reportID);
```

Comment:
@parasharrajat this is the change that solves the issue. When the IOU transaction comes in from pusher, we were previously calling `openReport`, even though the user wasn't actually opening the report, which would mark the report as read. 

I instead updated that to call `reconnectToReport` which fetches the report data without marking it as read.